data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -72,19 +72,38 @@ class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyEntityPayload
     this.vertices.filter(vertex => vertex._2 != null)
   }
 
-  /** Return entity based on ID at a timestamp
-   *
-   * @param entity    Edge or Vertex
-   * @param timestamp Insant
-   * @return Entity Object and HashMap
-   */
-  override def getEntity[T <: Entity](entity: T, timestamp: Instant): Option[(T, Attributes)] = ???
+  override def getEntity[T <: Entity](entity: T, timestamp: Instant): Option[(T, Attributes)] = {
+    entity match {
+      case _: VERTEX => getVertex(entity, timestamp)
+      case _: EDGE => getEdge(entity, timestamp)
+      case _ => None
+    }
+  }
+
+  def getVertex[T <: Entity](vertex: T, instant: Instant): Option[(T, Attributes)] = {
```

Comment:
Disse trenger ikke være generiske
```suggestion
  def getVertex(vertex: VERTEX, instant: Instant): Option[(VERTEX, Attributes)] = {
```

Reply:
You sure? Den klager på typemismatching osv.